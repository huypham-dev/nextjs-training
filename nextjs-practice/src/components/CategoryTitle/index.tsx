import { memo } from "react";

type Props = {
  category: string;
};

const CategoryTitle = ({ category }: Props) => (
  <div className="text-center px-6 py-12 mb-6 bg-gray-100 border-b">
    <h1 className="text-xl md:text-4xl pb-4">{category.toUpperCase()}</h1>
    <p className="leading-loose text-gray-dark">
      Catch up on the latest news and updates.
    </p>
  </div>
);

export default memo(CategoryTitle);
