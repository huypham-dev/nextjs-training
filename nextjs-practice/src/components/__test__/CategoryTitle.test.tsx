import { mount, shallow } from "enzyme";
import CategoryTitle from "../CategoryTitle";

describe("CategoryTitle", () => {
  it("Snapshot", () => {
    const component = shallow(<CategoryTitle category="Internet" />);

    expect(component).toMatchSnapshot();
  });

  it("Render correctly title", () => {
    const component = mount(<CategoryTitle category="Internet" />);
    const title = component.find("h1").text();
    const desc = component.find("p").text();

    expect(title).toBe("INTERNET");
    expect(desc).toBe("Catch up on the latest news and updates.");
  });
});
