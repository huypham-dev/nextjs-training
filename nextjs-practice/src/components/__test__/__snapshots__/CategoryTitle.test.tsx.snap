// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryTitle Snapshot 1`] = `
<div
  className="text-center px-6 py-12 mb-6 bg-gray-100 border-b"
>
  <h1
    className="text-xl md:text-4xl pb-4"
  >
    INTERNET
  </h1>
  <p
    className="leading-loose text-gray-dark"
  >
    Catch up on the latest news and updates.
  </p>
</div>
`;
